# HG changeset patch
# User Rik <rik@octave.org>
# Date 1348074315 25200
#      Wed Sep 19 10:05:15 2012 -0700
# Node ID a0fb36acbd0d78a92ec3b8dd85f98e7e2664379f
# Parent  15c435992d3cff2a9cc2511bc3813a9cf6cc70d1
build: Only use extra CPPFLAGS in the convenience libraries that need them.

* liboctave/Makefile.am: Remove XCPPFLAGS from liboctave_la_CPPFLAGS.

* liboctave/array/module.mk: Add SPARSE_XCPPFLAGS to convenience library
CPPFLAGS.

* liboctave/cruft/Makefile.am: Place all CPPFLAGS on one line.

* liboctave/numeric/module.mk: Add ARPACK_CPPFLAGS, FFTW_XCPPFLAGS,
SPARSE_XCPPFLAGS to convenience library CPPFLAGS.

* liboctave/util/module.mk: Add SPARSE_XCPPFLAGS to convenience library
CPPFLAGS.

diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -80,28 +80,21 @@ include numeric/module.mk
 include operators/module.mk
 include system/module.mk
 include util/module.mk
 
 ## liboctave merely collects a bunch of compiled convenience libraries.
 ## It has no source code itself.
 liboctave_la_SOURCES = 
 
-## FIXME: SPARSE, FFTW, and ARPACK flags should probably only be in the
-##        particular convenience library that needs them.  
-liboctave_la_CPPFLAGS = \
-  @OCTAVE_DLL_DEFS@ \
-  $(SPARSE_XCPPFLAGS) \
-  $(FFTW_XCPPFLAGS) \
-  $(ARPACK_CPPFLAGS) \
-  $(AM_CPPFLAGS)
+liboctave_la_CPPFLAGS = @OCTAVE_DLL_DEFS@ $(AM_CPPFLAGS)
 
 include link-deps.mk
 
-# Dummy C++ source to cause C++ linking.
+# Dummy C++ source to force C++ linking.
 nodist_EXTRA_liboctave_la_SOURCES = dummy.cc
 
 liboctave_la_LIBADD = \
   array/libarray.la \
   cruft/libcruft.la \
   cruft/ranlib/libranlib.la \
   numeric/libnumeric.la \
   operators/liboperators.la \
diff --git a/liboctave/array/module.mk b/liboctave/array/module.mk
--- a/liboctave/array/module.mk
+++ b/liboctave/array/module.mk
@@ -132,10 +132,12 @@ TEMPLATE_SRC += \
   array/MArray.cc \
   array/MDiagArray2.cc \
   array/MSparse.cc \
   array/Sparse.cc
 
 noinst_LTLIBRARIES += array/libarray.la
 
 array_libarray_la_SOURCES = $(ARRAY_SRC)
-array_libarray_la_CPPFLAGS = $(liboctave_la_CPPFLAGS)
+array_libarray_la_CPPFLAGS = \
+  $(liboctave_la_CPPFLAGS) \
+  $(SPARSE_XCPPFLAGS)
 
diff --git a/liboctave/cruft/Makefile.am b/liboctave/cruft/Makefile.am
--- a/liboctave/cruft/Makefile.am
+++ b/liboctave/cruft/Makefile.am
@@ -31,19 +31,17 @@ EXTRA_DIST = mkf77def.in
 noinst_LTLIBRARIES = libcruft.la
 
 libcruft_la_SOURCES =
 
 nodist_libcruft_la_SOURCES =
 
 libcruft_la_FFLAGS = $(F77_INTEGER_8_FLAG)
 
-libcruft_la_CPPFLAGS = \
-  @CRUFT_DLL_DEFS@ \
-  $(AM_CPPFLAGS)
+libcruft_la_CPPFLAGS = @CRUFT_DLL_DEFS@ $(AM_CPPFLAGS)
 
 libcruft_la_DEPENDENCIES = cruft.def
 
 octinclude_HEADERS =
 
 include amos/module.mk
 include blas-xtra/module.mk
 include daspk/module.mk
diff --git a/liboctave/numeric/module.mk b/liboctave/numeric/module.mk
--- a/liboctave/numeric/module.mk
+++ b/liboctave/numeric/module.mk
@@ -1,10 +1,10 @@
 EXTRA_DIST += \
-	numeric/module.mk \
+  numeric/module.mk \
   $(OPT_IN)
 
 OPT_INC = \
   numeric/DASPK-opts.h \
   numeric/DASRT-opts.h \
   numeric/DASSL-opts.h \
   numeric/LSODE-opts.h \
   numeric/Quad-opts.h
@@ -151,17 +151,17 @@ NUMERIC_SRC = \
   numeric/ODES.cc \
   numeric/Quad.cc \
   numeric/SparseCmplxCHOL.cc \
   numeric/SparseCmplxLU.cc \
   numeric/SparseCmplxQR.cc \
   numeric/SparsedbleCHOL.cc \
   numeric/SparsedbleLU.cc \
   numeric/SparseQR.cc \
-	$(NUMERIC_C_SRC)
+  $(NUMERIC_C_SRC)
 
 TEMPLATE_SRC += \
   numeric/base-lu.cc \
   numeric/base-qr.cc \
   numeric/bsxfun-defs.cc \
   numeric/eigs-base.cc \
   numeric/sparse-base-chol.cc \
   numeric/sparse-base-lu.cc \
@@ -171,10 +171,14 @@ TEMPLATE_SRC += \
 $(OPT_INC) : %.h : %.in $(top_srcdir)/build-aux/mk-opts.pl
 	@echo making $@ from $<
 	@$(PERL) $(top_srcdir)/build-aux/mk-opts.pl --opt-class-header $< > $@-t
 	mv $@-t $@
 
 noinst_LTLIBRARIES += numeric/libnumeric.la
 
 numeric_libnumeric_la_SOURCES = $(NUMERIC_SRC)
-numeric_libnumeric_la_CPPFLAGS = $(liboctave_la_CPPFLAGS)
+numeric_libnumeric_la_CPPFLAGS = \
+  $(liboctave_la_CPPFLAGS) \
+  $(ARPACK_CPPFLAGS) \
+  $(FFTW_XCPPFLAGS) \
+  $(SPARSE_XCPPFLAGS)
 
diff --git a/liboctave/util/module.mk b/liboctave/util/module.mk
--- a/liboctave/util/module.mk
+++ b/liboctave/util/module.mk
@@ -77,10 +77,12 @@ TEMPLATE_SRC += \
   util/oct-sort.cc
 
 OTHER_INC += \
   util/kpse.cc
 
 noinst_LTLIBRARIES += util/libutil.la
 
 util_libutil_la_SOURCES = $(UTIL_SRC)
-util_libutil_la_CPPFLAGS = $(liboctave_la_CPPFLAGS)
+util_libutil_la_CPPFLAGS = \
+  $(liboctave_la_CPPFLAGS) \
+  $(SPARSE_XCPPFLAGS)
 
